src/side-navigation/nav-item.js

```diff
@@ -49,7 +49,7 @@ export default class NavItem extends React.Component<NavItemPropsT> {
             }
           : {})}
       >
-        <NavItem {...sharedProps} {...itemProps}>
+        <NavItem item={item} {...sharedProps} {...itemProps}>
```

Comment:
`NavItem` is a styled element by default so passing the `item` prop here will pass it onto the `StyledNavItem` as an invalid attribute that will also result in a warning from React.